api/src/main/java/jakarta/persistence/PessimisticLockScope.java

```diff
@@ -11,53 +11,95 @@
-     * one-to-many relationships for which the target entity contains
-     * the foreign key) will not be locked by default.
+     * <p>This is the default behavior of pessimistic locking.
      */
     NORMAL,
 
     /**
-     * In addition to the locking behavior specified for {@link #NORMAL},
-     * element collections and relationships owned by the entity that
-     * are contained in join tables are locked if the property
-     * {@code jakarta.persistence.lock.scope} is specified with a value
-     * of {@code PessimisticLockScope#EXTENDED}. The state of entities
-     * referenced by such relationships is not locked (unless those
-     * entities are explicitly locked). Locking such a relationship or
-     * element collection generally locks only the rows in the join table
-     * or collection table for that relationship or collection. This means
-     * that phantoms are possible.
+     * <p>The persistence provider must lock the row or rows of:
+     * <ul>
+     * <li>the {@linkplain Table primary} and
+     *     {@linkplain SecondaryTable secondary} tables mapped by the
+     *     entity, including all rows in tables mapped by the entity
+     *     superclasses of the entity in the case of a
+     *     {@linkplain InheritanceType#JOINED joined} inheritance
+     *     hierarchy,
+     * <li>{@linkplain JoinTable join tables} mapped by associations
+     *     owned by the entity, and
+     * <li>{@linkplain CollectionTable collection tables} mapped by
+     *     collections belonging to the entity.
+     * </ul>
+     *
+     * <p>The pessimistic lock does not extend to rows of:
+     * <ul>
+     * <li>tables mapped by associated entities,
+     * <li>{@linkplain JoinTable join tables} mapped by associations
+     *     not owned by the entity.
+     * </ul>
+     */
+    EXTENDED,
+
+    /**
+     * <p>The persistence provider must lock the row or rows of:
+     * <ul>
+     * <li>the {@linkplain Table primary} and
+     *     {@linkplain SecondaryTable secondary} tables mapped by the
+     *     entity, including all rows in tables mapped by the entity
+     *     superclasses of the entity in the case of a
+     *     {@linkplain InheritanceType#JOINED joined} inheritance
+     *     hierarchy,
+     * <li>{@linkplain JoinTable join tables} mapped by associations
+     *     which were fetched as part of the operation which obtained
+     *     the pessimistic lock, along with the primary and secondary
+     *     tables mapped by the associated entities,
+     * <li>{@linkplain CollectionTable collection tables} mapped by
+     *     collections belonging to the entity which were fetched as
+     *     part of the operation which obtained the pessimistic lock.
+     * </ul>
+     *
+     * <p>The pessimistic lock does not extend to data which was not
+     * fetched as part of the operation which obtained the pessimistic
+     * lock.
+     *
+     * @since 4.0
      */
-    EXTENDED
+    CASCADED
```

Comment:
It sounds like you intend this to correlate to Hibernate's FETCH scope?  I think that is a better term.  "Cascaded" makes me think there is a new `CascadeType#LOCK` that we base the scope on.

Reply:
The current/legacy docs for EXTENDED were quite unclear and we missed locking owned join-tables in our original implementation. As our javadocs say, our impl is not entirely spec-compliant. We (Hibernate) could consider adding a fully spec compliant EXTENDED value/implementation if that is your point, I'm fine with that. But let's talk about that offline.